File: app/src/main/java/ua/com/zak/budgetswing/fragments/BaseFragment.java
package ua.com.zak.budgetswing.fragments;


import android.content.Context;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import butterknife.ButterKnife;
import ua.com.zak.budgetswing.ApplicationModelProvider;
import ua.com.zak.budgetswing.model.ApplicationModel;

/**
 * @author zak <zak@swingpulse.com>
 */
public abstract class BaseFragment extends Fragment {

    protected ApplicationModel mApplicationModel;

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        mApplicationModel = ((ApplicationModelProvider) context).getApplicationModel();
    }

    @Override
    public void onDetach() {
        super.onDetach();
        mApplicationModel = null;
    }

    @Nullable
    @Override
    public final View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View root = inflater.inflate(getLayoutId(), container);
        ButterKnife.bind(this, root);
        return root;
    }

    protected abstract int getLayoutId();
}

File: app/src/main/java/com/robotnec/budget/refactored/screens/BaseFragment.kt
package com.robotnec.budget.refactored.screens

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.robotnec.budget.refactored.di.helpers.Injectable
import dagger.android.support.AndroidSupportInjection

abstract class BaseFragment : Fragment() {
    abstract val layoutId: Int

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        if (javaClass.isAnnotationPresent(Injectable::class.java)) {
            AndroidSupportInjection.inject(this)
        }

        return inflater.inflate(layoutId, container, false)
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        initView(savedInstanceState)
    }

    abstract fun initView(savedInstanceState: Bundle?)
}
